src/daemon/abrt-action-save-package-data.c

```diff
@@ -69,67 +69,66 @@ static GList *settings_Interpreters = NULL;
+        settings_bOpenGPGCheck = libreport_string_to_bool((char *)value);
+        g_hash_table_remove(settings, "OpenGPGCheck");
     }
 
-    value = libreport_get_map_string_item_or_NULL(settings, "BlackList");
+    value = g_hash_table_lookup(settings, "BlackList");
     if (value)
     {
-        settings_setBlackListedPkgs = libreport_parse_delimited_list(value, ",");
-        libreport_remove_map_string_item(settings, "BlackList");
+        settings_setBlackListedPkgs = libreport_parse_delimited_list((char *)value, ",");
+        g_hash_table_remove(settings, "BlackList");
     }
     else
         settings_setBlackListedPkgs = libreport_parse_delimited_list(DEFAULT_BLACKLISTED_PKGS, ",");
 
-    value = libreport_get_map_string_item_or_NULL(settings, "BlackListedPaths");
+    value = g_hash_table_lookup(settings, "BlackListedPaths");
     if (value)
     {
-        settings_setBlackListedPaths = libreport_parse_delimited_list(value, ",");
-        libreport_remove_map_string_item(settings, "BlackListedPaths");
+        settings_setBlackListedPaths = libreport_parse_delimited_list((char *)value, ",");
+        g_hash_table_remove(settings, "BlackListedPaths");
     }
     else
         settings_setBlackListedPaths = libreport_parse_delimited_list(DEFAULT_BLACKLISTED_PATHS, ",");
 
-    value = libreport_get_map_string_item_or_NULL(settings, "ProcessUnpackaged");
+    value = g_hash_table_lookup(settings, "ProcessUnpackaged");
     if (value)
     {
-        settings_bProcessUnpackaged = libreport_string_to_bool(value);
-        libreport_remove_map_string_item(settings, "ProcessUnpackaged");
+        settings_bProcessUnpackaged = libreport_string_to_bool((char *)value);
+        g_hash_table_remove(settings, "ProcessUnpackaged");
     }
 
-    value = libreport_get_map_string_item_or_NULL(settings, "Interpreters");
+    value = g_hash_table_lookup(settings, "Interpreters");
     if (value)
     {
-        settings_Interpreters = libreport_parse_delimited_list(value, ",");
-        libreport_remove_map_string_item(settings, "Interpreters");
+        settings_Interpreters = libreport_parse_delimited_list((char *)value, ",");
+        g_hash_table_remove(settings, "Interpreters");
     }
 
     map_string_iter_t iter;
-    const char *name;
-    /*char *value; - already declared */
-    libreport_init_map_string_iter(&iter, settings);
-    while (libreport_next_map_string_iter(&iter, &name, &value))
+    gpointer name;
+    g_hash_table_iter_init(&iter, settings);
+    while (g_hash_table_iter_next(&iter, &name, &value))
     {
-        error_msg("Unrecognized variable '%s' in '%s'", name, conf_filename);
+        error_msg("Unrecognized variable '%s' in '%s'", (char *)name, conf_filename);
     }
 }
 
 static void load_gpg_keys(void)
 {
-    map_string_t *settings = libreport_new_map_string();
+    map_string_t *settings = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, g_free);
```

Comment:
This variable doesn't seem to be freed anywhere inside the function. Another candidate for autoptr.